src/main/java/build/buildfarm/cas/cfc/CASFileCache.java

```diff
@@ -1974,13 +1977,41 @@ private void putDirectoryFiles(
   }
 
   private void fetchDirectory(
-      Path path,
+      Path rootPath,
       Digest digest,
       Map<Digest, Directory> directoriesIndex,
       ImmutableList.Builder<String> inputsBuilder,
       ImmutableList.Builder<ListenableFuture<Path>> putFutures,
       ExecutorService service)
       throws IOException, InterruptedException {
+    Stack<Pair<Path, Directory>> stack = new Stack<>();
```

Comment:
Why not use Map.Entry (with instantiations via AbstractMap.SimpleEntry) rather than incorporate another apache dependency?

Reply:
good idea!  fixed.